core/src/main/kotlin/me/frmr/github/repopolicy/core/PolicyEngine.kt

```diff
@@ -132,12 +141,23 @@ class PolicyEngine(val policy: PolicyDescription, val logging: Boolean) {
     return policy.rules.flatMap { rule ->
       val repos = findMatchingRepos(rule.subject) ?: emptyList()
       repos.flatMap { repo ->
+        // refresh app installation token if necessary
+        if (this.appInstallationTokenTtl > 0 && this.appInstallationTokenTtl < System.currentTimeMillis()) {
+          initGithubClient()
```

Comment:
There's got to be a better way to handle this than just copy pastaing this block around but it's not occurring to me at the moment

Reply:
I'll take a look at cleaning this up a little bit.